app/front-end/src/features/editor/components/fileTreeView/fileTreeItem/fileTreeItemContextMenuTextfieldDialog.tsx

```diff
@@ -0,0 +1,301 @@
+        if (item.fileType !== FileTypes.FOLDER) return item.label.match(/[^.]+$/)?.[0] || FileTypes.CSV;
+        return '';
+      default:
+        return '';
+    }
+  });
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (open) {
+      setValue(() => {
+        switch (action) {
+          case FileTreeItemContextMenuActions.RENAME:
+            if (item.fileType !== FileTypes.FOLDER) return item.label.match(/^(.*)(?=\.[^.]+$)/)?.[0] || '';
+            return item.label;
+          default:
+            return '';
+        }
+      });
+      setFileExtension(() => {
+        switch (action) {
+          case FileTreeItemContextMenuActions.NEW_FILE:
+            return FileTypes.CSV;
+          case FileTreeItemContextMenuActions.NEW_FOLDER:
+            return '';
+          case FileTreeItemContextMenuActions.RENAME:
+            if (item.fileType !== FileTypes.FOLDER) return item.label.match(/[^.]+$/)?.[0] || FileTypes.CSV;
+            return '';
+          default:
+            return '';
+        }
+      });
+      setError(null);
+    }
+  }, [open, item, action]);
+
+  const validateInput = (input: string, fileExtension: string) => {
+    const parentPath = item.id.match(/^(.*)(?=\/[^/]*$)/)?.[0] || '';
+    const path = () => {
+      if (action === FileTreeItemContextMenuActions.RENAME) {
+        if (parentPath === '') return input;
+        return parentPath + '/' + input;
+      } else {
+        if (item.id === '') return input;
+        return item.id + '/' + input;
+      }
+    };
+
+    // Check if file already exists
+    switch (action) {
+      case FileTreeItemContextMenuActions.NEW_FILE:
+        if (doesFileExist(fileTreeViewItems || [], path() + '.' + fileExtension)) return 'This name already exists';
+        break;
+      case FileTreeItemContextMenuActions.NEW_FOLDER:
+        if (doesFileExist(fileTreeViewItems || [], path())) return 'This name already exists';
+        break;
+      case FileTreeItemContextMenuActions.RENAME:
+        if (fileExtension === '') {
+          if (doesFileExist(fileTreeViewItems || [], path())) return 'This name already exists';
+        } else {
+          if (doesFileExist(fileTreeViewItems || [], path() + '.' + fileExtension)) return 'This name already exists';
+        }
+        break;
+      default:
+        break;
+    }
+
+    // Check if the input is empty
+    if (!input) {
+      return 'Input cannot be empty';
+    }
+
+    // Check if the input exceeds the length limit
+    if (input.length > 50) {
+      return 'Input must be less than 50 characters';
+    }
+
+    // Check for forbidden characters
+    if (input.includes('\0')) {
+      return 'Input contains a forbidden null character';
+    }
+
+    if (input.includes('/')) {
+      return 'Input cannot contain a forward slash (/)';
+    }
+
+    if (/[*?[\]]/.test(input)) {
+      return 'Input contains forbidden glob characters (*, ?, [, ])';
```

Comment:
We can remove the 'glob' word because I don't think many people would know it's meaning, "...forbidden characters..." should be enough

Reply:
Changed!